lib/KListWithOverflow.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+
+  <div
+    ref="listWrapper"
+    class="list-wrapper"
+    :style="appearanceOverrides"
+  >
+    <div
+      ref="list"
+      class="list"
+    >
+      <template v-for="(item) in items">
+        <!-- @slot Slot for rendering divider items -->
+        <slot
+          v-if="isDivider(item)"
+          name="divider"
+          :divider="item"
+        >
+        </slot>
+        <!-- @slot Slot responsible of rendering each item in the visible list -->
+        <slot
+          v-else
+          name="item"
+          :item="item"
+        ></slot>
+      </template>
+    </div>
+    <div
+      ref="moreButtonWrapper"
+      class="more-button-wrapper"
+    >
+      <!-- @slot Slot responsible of rendering the "see more" button. This slot receives as prop a list `overflowItems` with items that dont fit into the visible list.-->
+      <slot
+        v-if="isMoreButtonVisible"
+        name="more"
+        :overflowItems="overflowItems"
+      ></slot>
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  import throttle from 'lodash/throttle';
```

Comment:
I ran some tests with `frame-throttle` and it seems that it excecute the method much more times than if we throttled it with 100ms, in both scenarios, when I manually resize the window, and when there is an animation:

using  `frame-throttle`:

https://github.com/learningequality/kolibri-design-system/assets/51239030/425ef33e-2869-405d-8211-e7d9a58b558a


using  `lodash/throttle` with 100ms

https://github.com/learningequality/kolibri-design-system/assets/51239030/bf8cfbac-62b6-40e0-85e2-57e876eae29d

Reply:
@AlexVelezLl The main benefit of `frame-throttle` is not lesser frequency but rather that it can choose the moment to repaint in a more clever way, see https://github.com/learningequality/kolibri-design-system/pull/469#discussion_r1358711535. The metrics here would be 'how many force repaints' rather than 'how often'. It seems particularly useful to prevent from extra repaints that will be happening to some extent via this component. I don't know exactly how these two techniques compare in this particular scenario and think we'd need to measure performance for larger amounts of instances to be certain. So it's a minor note based largely on intuition. You're welcome to take any direction ;) Thanks for looking into it.

Reply:
Thanks! I find this topic very interesting, and I have run some other tests isolating the component and testing with multiple instances, and I don't see much difference in the repainting of both. Here are some tests:

Using `frame-throttle`:

https://github.com/learningequality/kolibri-design-system/assets/51239030/dbe2d1e6-d63e-41b0-a2cb-bc825f37a9fc

Using `lodash/throttle` with 100ms:

https://github.com/learningequality/kolibri-design-system/assets/51239030/884be1f0-d280-45cb-8eb6-ffb6ba7d65ca

But the script time taken for the same example differs drastically, and using `frame-throttle` is 3x expensive:

Using `frame-throttle`:

![image](https://github.com/learningequality/kolibri-design-system/assets/51239030/2dc808a9-3d45-400a-91e3-6a3664a0e2f1)

Using `lodash/throttle` with 100ms:

![image](https://github.com/learningequality/kolibri-design-system/assets/51239030/6a7b64fe-112c-4a44-af6b-d81b58b162d1)

I see that, indeed, the painting takes less time, but compared to the difference in the script time, I think is ok preferring the `lodash/throttle` for this example.

Reply:
(At the end I just ended adding the `frame-throttle` in the `_useKResponsiveElement` as I think that makes sense and we follow the pattern of the `useKWindowDimensions`, but kept the regular throttle with 100ms in the elementWidth watch in KListWithOverflow) :open_hands:.

Reply:
@AlexVelezLl thank you for looking into it in detail